x-pack/osquerybeat/ext/osquery-extension/pkg/amcache/registry/registry.go

```diff
@@ -0,0 +1,249 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package registry
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/forensicanalysis/fslib"
+	"github.com/forensicanalysis/fslib/systemfs"
+	"www.velocidex.com/golang/regparser"
+
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/logger"
+)
+
+// ensureLogger ensures that a logger is provided. If no logger is provided,
+// a new logger is created and returned. A simple check to ensure
+// that the logger is not nil.
+func ensureLogger(log *logger.Logger) *logger.Logger {
+	if log == nil {
+		log = logger.New(os.Stderr, false)
+	}
+	return log
+}
+
+// getFileContents reads the contents of a file and returns it as a byte slice.
+// If the file is not readable, it will attempt to read it using a low level read
+// using fslib.
+func getFileContents(filePath string, log *logger.Logger) ([]byte, error) {
+	log = ensureLogger(log)
+	content, err := os.ReadFile(filePath)
+	if err == nil {
+		return content, nil
+	}
+	log.Infof("failed to read %s, falling back to low level read", filePath)
+
+	// fallback to a low level read using fslib
+	sourceFS, err := systemfs.New()
+	if err != nil {
+		log.Errorf("failed to open file %s: %s", filePath, err.Error())
+		return nil, err
+	}
```

Comment:
since this does not seem to be parameterized, could we store the sourceFS and reuse it? or at least make it valid for the same period of time as the gInstance? Is that something that would make sense or be better in any case?